### ISSUE TITLE
Feature: JetBrains/TeamCity/RubyMine Formatted Output

### ISSUE THREAD

**Issue by dacat** [rodjek/puppet-lint#702]
## TL;DR
Make puppet-lint output the results so RubyMine can parse it and be consumed by the RubyMine test runner 
* Feature branch [here](https://github.com/dacat/puppet-lint/tree/feature/teamcity-formatter).  
*  Travisci build [here](https://travis-ci.org/dacat/puppet-lint).
### Profit!
>![rubymine-event-log-and-test-windows](https://cloud.githubusercontent.com/assets/37885/25722283/373231f4-30d9-11e7-8a3b-ea1d404664e4.PNG)

## How This Came About
RubyMine introduced puppet support recently.  Support for linting was lacking.  My initial stab at making the output from puppet-lint consumable was capturing the I/O then parsing the individual lines and formatting them in a way RubyMine liked.  It worked great but I continued to scratch the itch and wanted puppet-lint to support the format out of the box.

## Solution
- Added a new option: --teamcity-formatter
- Added specs to cover the new option
- Copied over TeamCity message format support files from rubymine_helpers supplied with RubyMine
- Added RakeTask support for the new option.
- Added the TeamCity::PuppetLintFormatter class

I have a fork off of master and started a feature branch [here](https://github.com/dacat/puppet-lint/tree/feature/teamcity-formatter).  The feature has specs covering it as well as a travisci build [here](https://travis-ci.org/dacat/puppet-lint).

### Normal Output
![normal-output-sample](https://cloud.githubusercontent.com/assets/37885/25722926/8d847b1e-30db-11e7-961f-f0fba5906617.PNG)
### TeamCity/RubyMine/JetBrains Formatted Output
![teamcity-formatter-sample](https://cloud.githubusercontent.com/assets/37885/25722956/a73a24fa-30db-11e7-88c1-62ad63fbd2c9.PNG)

### Configure RubyMine to Consume the Output
1. Add the PuppetLint RakeTask 
>![raketask-example](https://cloud.githubusercontent.com/assets/37885/25723903/81cbf028-30df-11e7-8097-a8e110a1b117.PNG)
2. Add Run configuration and check the Rspec output 
>![rubymine-run-config-to-use-puppet-lint](https://cloud.githubusercontent.com/assets/37885/25723909/8afd809e-30df-11e7-83af-ff0d0dc8ffb2.PNG)
3. Profit!
>![rubymine-event-log-and-test-windows](https://cloud.githubusercontent.com/assets/37885/25723921/966c05fe-30df-11e7-8e4b-f6050f6443dc.PNG)

## Next Steps
1. Code Review with main developers.
    * Let me know if its worthwhile for this to be included in puppet-lint if so, lets have a chat and code review.
1. More documentation. 
    * More clarification as to the individual steps the formatter uses. 
    * More clarification as the how the  JetBrains portions work
1. Testing with TeamCity
1. Include the tests that are running for each file in the output so the user sees things that passed.
   * currently the output says "Tests Failed: 0 passed, ### failed"
   * I would like it to say "Tests Failed: ### passed, ### failed, ### ignored" where the number of tests passed equals the number of files + the number of checks executed and the the number of tests ignored is the number of disabled checks.
1. Remove the dependency on the copied code
    * Basically strip out all the pieces not needed for the output and incorporate it in to the TeamCity::PuppetLintFormatter class.
